frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.effects.ts

```diff
@@ -199,19 +204,59 @@ export class FeatureFlagsEffects {
 
   fetchFeatureFlagById$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(FeatureFlagsActions.actionFetchFeatureFlagById),
+      ofType(featureFlagsActions.actionFetchFeatureFlagById),
       map((action) => action.featureFlagId),
       filter((featureFlagId) => !!featureFlagId),
       switchMap((featureFlagId) =>
         this.featureFlagsDataService.fetchFeatureFlagById(featureFlagId).pipe(
           map((data: FeatureFlag) => {
-            return FeatureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
+            return featureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
           }),
-          catchError(() => [FeatureFlagsActions.actionFetchFeatureFlagByIdFailure()])
+          catchError(() => [featureFlagsActions.actionFetchFeatureFlagByIdFailure()])
         )
       )
     )
   );
 
+  emailFeatureFlagData$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(featureFlagsActions.actionEmailFeatureFlagData),
+      map((action) => ({ featureFlagId: action.featureFlagId })),
+      withLatestFrom(this.store$.pipe(select(selectCurrentUser))),
+      filter(([{ featureFlagId }, { email }]) => !!featureFlagId && !!email),
+      switchMap(([{ featureFlagId }, { email }]) =>
+        this.featureFlagsDataService.emailFeatureFlagData(featureFlagId, email).pipe(
+          map(() => {
+            this.notificationService.showSuccess(`Email will be sent to ${email}`);
+            return featureFlagsActions.actionEmailFeatureFlagDataSuccess();
+          }),
+          catchError(() => [featureFlagsActions.actionEmailFeatureFlagDataFailure()])
+        )
+      )
+    )
+  );
+
+  exportFeatureFlagsDesign$ = createEffect(() =>
+  this.actions$.pipe(
+    ofType(featureFlagsActions.actionExportFeatureFlagDesign),
+    withLatestFrom(this.store$.pipe(select(selectFeatureFlagIds))),
```

Comment:
withLatestFrom(this.store$.pipe(select(selectFeatureFlagIds))) can be removed, will be able to just pass in single id @ `this.featureFlagsDataService.exportFeatureFlagsDesign()`